### PR TITLE
Reflection_oM: Migrate objects to Base_oM - Removing Reflection_oM

### LINKED ISSUE #1338: Reflection_oM: Move objects to Base_oM
In line with:

https://github.com/BHoM/admin/issues/17
https://github.com/BHoM/BHoM_Engine/issues/1645
https://github.com/BHoM/BHoM_Engine/issues/2671

Move the objects within Reflection_oM into Base_oM where the objects do not allow the code to ask questions about itself. This will result in all objects moving. See diagram below highlighting the new Base_oM after this move.

image

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
### Issues addressed by this PR
Fixes #1338 

### Changelog
 - All Reflection_oM objects moved to Base_oM
 - Removal of `DeprecatedAttribute` (deprecated itself in v3.2)
 - Removal of Reflection_oM project

### Additional comments
Please read through the issues associated linked via #1338 prior to reviewing.

### CODE REVIEW

**Comment by adecler** [comment]
There a still a few classes that use the Deprecated attribute:

image

**Comment by IsakNaslundBh** [comment]
> There a still a few classes that use the Deprecated attribute:
> 
> image

Regarding the Grids, can we put a `ToBeDeleted` attribute on them temporarily?

Was planning to properly version them when I close out https://github.com/BHoM/BHoM/issues/756 finally by migrating grids and levels from Geometry_oM to Spatial_oM. When doing so planning to version the architectural grids to the spatial ones as well in one go and then simply delete them.

File: BHoM/UnderlyingType.cs [hunk 1]
```diff
@@ -20,14 +20,13 @@
...
 
-using BH.oM.Base;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
 
-namespace BH.oM.Reflection
+namespace BH.oM.Base
```
Comment (IsakNaslundBh) [inline BHoM/UnderlyingType.cs:29]: This class does not make full sense to me in the basefolder next to the BHoMObejct. Would suggest moving it to a subfolder and potentially sub-namespace. `System` as a suggestion, but not sure that is ideal. Open for other ideas.

**Review by IsakNaslundBh — COMMENTED** [review]
Suggesting moving UnderlyingType down to a subfolder. Appart from that and what @adecler already mentioned, things are generally looking good.

Small tweak to description on `UnitTest` `Method` Property might be waranted if the MethodList is moved to the the base BHoM_Engine.

File: BHoM/UnderlyingType.cs (hunk 1 above)
Comment (IsakNaslundBh) [reply BHoM/UnderlyingType.cs:29]: Happy for this to be handled in a later set of PRs as that only requires alignment over a few set of repos, not the whole of BHoM

**Review by IsakNaslundBh — APPROVED** [review]
Generally happy with this. Think the comment on UnderlyingType can be handled seprately in a more calm set of PRs.

For versioning of objects we can rely on the Bot. If that is passing, this looks good to me.

**Comment by FraserGreenroyd** [comment]
@BHoMBot check installer

Let's do this...

**Comment by FraserGreenroyd** [comment]
@BHoMBot check installer

Let's try that again...

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check serialisation
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: core, code-compliance, serialisation, ready-to-merge

File: BHoM/UnderlyingType.cs (hunk 1 above)
Comment (FraserGreenroyd) [reply BHoM/UnderlyingType.cs:29]: Issue has been raised [here](https://github.com/BHoM/BHoM/issues/1346)

**Pull request merged**